diff --git a/src/torchcodec/_core/CMakeLists.txt b/src/torchcodec/_core/CMakeLists.txt
index 1bcb5f8..e3a5e1a 100644
--- a/src/torchcodec/_core/CMakeLists.txt
+++ b/src/torchcodec/_core/CMakeLists.txt
@@ -17,9 +17,18 @@ if(SKBUILD_STATE STREQUAL "wheel"
 endif()
 
 set(PYBIND11_FINDPYTHON ON)
+
+if(NOT pybind11_DIR)
+    execute_process(
+        COMMAND python3 -m pybind11 --cmakedir
+        OUTPUT_VARIABLE pybind11_DIR
+        OUTPUT_STRIP_TRAILING_WHITESPACE
+    )
+endif()
+
 find_package(pybind11 REQUIRED)
 find_package(Torch REQUIRED)
-find_package(Python3 COMPONENTS Development Interpreter)
+find_package(Python3 COMPONENTS Development.Module Interpreter)
 
 if(APPLE)
     # Some CI runners ship a Mono.framework which contains headers for libjpeg,
diff --git a/src/torchcodec/_core/fetch_and_expose_non_gpl_ffmpeg_libs.cmake b/src/torchcodec/_core/fetch_and_expose_non_gpl_ffmpeg_libs.cmake
index af9df33..d9678cd 100644
--- a/src/torchcodec/_core/fetch_and_expose_non_gpl_ffmpeg_libs.cmake
+++ b/src/torchcodec/_core/fetch_and_expose_non_gpl_ffmpeg_libs.cmake
@@ -11,7 +11,7 @@ include(FetchContent)
 
 set(
     base_url
-    https://pytorch.s3.amazonaws.com/torchcodec/ffmpeg/2025-03-14
+    https://github.com/loong64/torchcodec/raw/refs/heads/ffmpeg/2025-03-14
 )
 
 if (LINUX)
@@ -45,6 +45,36 @@ if (LINUX)
             f9_sha256
             e68d3a5a17705740f0f8cf69a92ab65ae47c9e17eb3969b17966bd11ab12b382
         )
+    elseif (CMAKE_SYSTEM_PROCESSOR MATCHES "loongarch64")
+        set(
+            platform_url
+            ${base_url}/linux_loongarch64
+        )
+
+        set(
+            f4_sha256
+            ef11e70e29343e4d4ecc6225cd1633b0050e169b61a7bfbeab19fd52655f74d1
+        )
+        set(
+            f5_sha256
+            ed5cf61ce779527f2786485285c409a438796ebcdc651e67a31e8c4fba817118
+        )
+        set(
+            f6_sha256
+            6d7bb0d674e4e3edc16008df59f98025d3eac00159c5fc9deaa184db82de565e
+        )
+        set(
+            f7_sha256
+            f68da8365a3f71f46a36c70060ae2c58e171075f6807a2e5b13a1c13195abb8e
+        )
+        set(
+            f8_sha256
+            049580298a415d63d321471f86ef541bdc7693fe8bb22cf69ead4f7314a1f8d7
+        )
+        set(
+            f9_sha256
+            7bd8b872a8d718a6fd245656b51d634f764dfcf173eee77c801914196ea91dc8
+        )
     elseif (LINUX)  # assume x86_64
         set(
             platform_url
diff --git a/src/torchcodec/_core/fetch_avif_from_s3.cmake b/src/torchcodec/_core/fetch_avif_from_s3.cmake
index 4b1a241..e2e3595 100644
--- a/src/torchcodec/_core/fetch_avif_from_s3.cmake
+++ b/src/torchcodec/_core/fetch_avif_from_s3.cmake
@@ -17,13 +17,16 @@ set(libavif_version "1.4.2")
 
 set(
     base_url
-    https://pytorch.s3.amazonaws.com/torchcodec/libavif/2026-07-22
+    https://github.com/loong64/torchcodec/raw/refs/heads/libavif/2026-07-22
 )
 
 if (LINUX)
     if (CMAKE_SYSTEM_PROCESSOR MATCHES "aarch64|arm64|ARM64")
         set(platform_url ${base_url}/linux_aarch64)
         set(avif_sha256 9a377d3183bfabdc2e5f2e2f3969d8c3d605de56640b695832827aa7df8a8cce)
+    elseif (CMAKE_SYSTEM_PROCESSOR MATCHES "loongarch64")
+        set(platform_url ${base_url}/linux_loongarch64)
+        set(avif_sha256 d1ac190d2a202ad5fedc2ec2e510ab5db6d2b5d59efee4e56064ee2db5811ee4)
     else()  # assume x86_64
         set(platform_url ${base_url}/linux_x86_64)
         set(avif_sha256 b814cbb53ede0b2e32f4907116f466cd68a789283ba8be149d919fb33d336300)
